File: MenzaMateBackEnd/MenzaMateBackend/MenzaMate.Data/Entities/Survey.cs
using MenzaMate.Data.Static;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;

namespace MenzaMate.Data.Entities
{
    public class Survey
    {
        public int SurveyId { get; set; }
        public string SurveyName { get; set; }
        public string SurveyDescription { get; set; }
        public DateTime SurveyDate { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public ICollection<Question> Questions { get; set; } = new List<Question>();


    }
    public class SurveyConfigurationBuilder : IEntityTypeConfiguration<Survey>
    {
        public void Configure(EntityTypeBuilder<Survey> builder)
        {
            builder.HasKey(x => x.SurveyId);

            builder
                .Property(x => x.SurveyName)
                .IsRequired()
                .HasMaxLength(DataValidationConstants.LowMaxLength);

            builder
                .Property(x => x.SurveyDescription)
                .IsRequired();
        }
    }
}

File: MenzaMateFrontEnd/ws/src/main/java/models/Survey.kt
package models


data class Survey(
    val surveyId: Int,
    val surveyName: String?,
    val surveyDescription: String?,
    val surveyDate: String?,
    val userId: Int,
    val questions: List<Question>?,
    val responses: List<SurveyResponse>?
)


data class Question(
    val questionId: Int,
    val questionText: String?,
    val surveyId: Int,
    val answers: List<String>?,
    val userAnswers: List<UserResponse>?
)


data class SurveyResponse(
    val surveyId: Int,
    val userId: Int,
    val responses: List<UserSurveyResponse>
)

data class UserSurveyResponse(
    val questionId: Int,
    val answer: String
)
